src/plots/cartesian/set_convert.js

```diff
@@ -178,8 +179,8 @@ module.exports = function setConvert(ax, fullLayout) {
         if(isNumeric(v)) return +v;
     }
 
-    function getRangePosition(v) {
-        return isNumeric(v) ? +v : getCategoryIndex(v);
+    function getRangePositionForCategory(v) {
+        return (typeof v !== 'string') ? v : getCategoryIndex(v);
```

Comment:
Could you please elaborate why using `isNumeric` here is not possible?

Reply:
`isNumeric` returns `true` for `isNumeric('1')` which we don't want. We want numeric string categories to have to use the indices.